### PR TITLE
Docs: Add automated theme.json reference documentation

### LINKED ISSUE #36278: Documentation: Generate theme.json metadata documentation from schema file
## What problem does this address?

With the theme.json file now a part of the Gutenberg repository (https://github.com/WordPress/gutenberg/pull/35998) we can take advantage of this file and use it to auto-generate theme.json reference documentation.

The [current theme.json documentation page](https://github.com/WordPress/gutenberg/blob/trunk/docs/how-to-guides/themes/theme-json.md) includes a fair amount of explanation on using the theme.json that makes it hard to use as a reference.

## What is your proposed solution?

My proposal is to generate a new API reference document in the [reference-guides section](https://github.com/WordPress/gutenberg/tree/trunk/docs/reference-guides) that includes information from the schema file. This will make it easier when adding new settings and styles, because just the schema would need to be updated and it would self-document the new features.

The current documentation in the how-to-guides section could then be slimmed down and focus on explaining how to use the theme.json and not have to be updated to try and track every change. 

cc: @oandregal - since you've been greatly involved in theme.json (and automated docs)

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Uses the theme.json schema document to generate a reference guide for settings and styles.

Fixes: #36278

Note: This is a stacked PR on top of #37558 that adds npm-run-all command, that should be merged first.

## How has this been tested?

See a [rendered view of the documentation](https://github.com/WordPress/gutenberg/blob/docs/36278-gen-theme-ref/docs/reference-guides/theme-json-reference.md).

## Types of changes

Documentation update.

- Generates a new reference document for theme.json
- Adds the script to the docs:build process

### CODE REVIEW

**Comment by carolinan** [comment]
Not a code review -the rendered view looks great. 
Wishlist: Include what tools `appearanceTools` enables.

**Comment by mkaz** [comment]
@carolinan Good suggestion, I updated to include the tools that are enabled.

@ajlende I added a `markdownDescription` field and use that if present, this mimicks the [tsconfig schema](http://json.schemastore.org/tsconfig) you showed me. This allows for better formatting for the docs. 👍

File: schemas/json/theme.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
 		"settingsProperties": {
 			"properties": {
 				"appearanceTools": {
-					"description": "Setting that enables ui tools.",
+					"description": "Setting that enables enable UI tools.",
+					"markdownDescription": "Setting that enables the following UI tools:\n\n- border: color, radius, style, width\n- color: link\n- spacing: blockGap, margin, padding\n- typography: lineHeight",
```
Comment (ajlende) [inline schemas/json/theme.json]: I think the additional information here can be added to `description` for now. We've been sticking strictly to the draft-04 spec to ensure the schema will work with the widest variety of tools.

`markdownDescription` was added by Microsoft specifically to support Visual Studio Code (as far as I'm aware) and isn't a valid draft-04 property.

I'd like to see a little bit of research on tool usage for theme developers and on how a few more of the tools handle invalid properties before beginning to include them.

**Review by ajlende — COMMENTED** [review]
Seems like a solid foundation to build upon!

I'd eventually like to see more descriptions for properties included. Things like formatting for color strings are important and aren't included yet.

Examples are another thing that could be useful. Draft 6 added the `examples` property that we could leverage; however, like `markdownDescription` I'd want to do a little research before moving away from draft-04.

File: schemas/json/theme.json (hunk 1 above)
Comment (mkaz) [reply schemas/json/theme.json]: ok, I just moved the same richer text to the description field. 👍

**Review by ajlende — APPROVED** [review]
Looks good to me!

**Pull request merged**